### PR TITLE
docs: cleanup MessageCreateOptions and MessageReplyOptions

### LINKED ISSUE #9209: `MessageReplyOptions` references non-existent type definition `BaseMessageCreateOptions`
### Which package is this bug report for?

discord.js

### Issue description

As seen over at [`MessageReplyOptions`](https://discord.js.org/#/docs/discord.js/main/typedef/MessageReplyOptions), this lists `BaseMessageCreateOptions` which does not appear to have a link to anything.

### Code sample

_No response_

### Package version

N/A

### Node.js version

N/A

### Operating system

_No response_

### Priority this issue should have

Low (slightly annoying)

### Which partials do you have configured?

Not applicable (subpackage bug)

### Which gateway intents are you subscribing to?

Not applicable (subpackage bug)

### I have tested this issue on a development release

4367ab930227048868db3ed8437f6c4507ff32e1

### PR DESCRIPTION
**Pull request by BoogeyMan24**
**Please describe the changes this PR makes and why it should be merged:**
Changed `BaseMessageCreateOptions` to `BaseMessageOptions` and added `tts`, `nonce`, `stickers`, and `flags` as well. All near line 822 in `discordjs/src/structures/Messages.js`.

Hopefully this fixes #9209.

Edit: As suggested by Jiralite and for deduplication purposes, `BaseMessageCreateOptions` has been created to contain the the shared data of `MessageCreateOptions` and `MessageReplyOptions`. `MessageCreateOptions` and `MessageReplyOptions` have been updated to extend `BaseMessageCreateOptions` and have their respective special properties added `reply`, and `failIfNotExists`.

I probably may have missed something, so if I did just send a comment and I will change it.

**Status and versioning classification:**
There are no code changes
This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by jaw0r3k** [comment]
But it already have pr?

**Comment by BoogeyMan24** [comment]
I deleted stickers because MessageCreateOptions already has stickers.

**Review by Jiralite — CHANGES_REQUESTED** [review]
Extending `MessageCreateOptions` isn't correct. It now contains irrelevant properties such as `reply`, etc.

**Comment by BoogeyMan24** [comment]
Should I just then extend `BaseMessageOptions` and add everything except `reply`

**Comment by Jiralite** [comment]
For purposes of deduplication, I would:

1. Create `BaseMessageCreateOptions` which has everything `MessageCreateOptions` has right now except `reply`.
2. Have `MessageCreateOptions` extend `BaseMessageCreateOptions` and add `reply`.
3. Create `MessageReplyOptions` and have it extend `BaseMessageCreateOptions` and add `failIfNotExists`.

**Comment by BoogeyMan24** [comment]
I added the changes for deduplication.

File: packages/discord.js/src/structures/interfaces/TextBasedChannel.js [hunk 1]
```diff
@@ -72,18 +72,23 @@ class TextBasedChannel {
    * @property {boolean} [failIfNotExists=this.client.options.failIfNotExists] Whether to error if the referenced
    * message does not exist (creates a standard message in this case when false)
    */
-
+
```
Comment (jaw0r3k) [inline packages/discord.js/src/structures/interfaces/TextBasedChannel.js:75]:
```suggestion

```

**Review by Jiralite — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**